### PR TITLE
NH-3578 - Subcriteria.UniqueResult<T> for value types should return default(T), same as CriteriaImpl.UniqueResult<T>

### PR DESCRIPTION
**Pull request by hazzik**
rework of #243

### CODE REVIEW

**Comment by hazzik** [comment]
Fixed compilation issue. Should be good now.

File: src/.gitignore [hunk 1]
```diff
@@ -33,4 +33,5 @@ _ReSharper*/
 SharedAssemblyInfo.*
 hibernate.cfg.xml
 .vs/
-packages/
\ No newline at end of file
+packages/
+.idea/
```
Comment (fredericDelaporte) [inline src/.gitignore]: IntelliJ? And maybe unintended, I do not see how it relates to this PR.

**Comment by hazzik** [comment]
Rider. Not related:)

**Comment by hazzik** [comment]
Probably was captured during git amendment

**Comment by fredericDelaporte** [comment]
Probably no harm to have it and more convenient. For who is using this IDE.

VS is terribly slow on NHibernate solution since project format change. Slow on load, freezes many seconds frequently after file changes by an external process, "reloading" changes for compiling during half a minute... Maybe I should check Rider.

**Pull request merged**